shared/src/main/scala/tastyquery/reader/TreeUnpickler.scala

```diff
@@ -558,7 +557,7 @@ class TreeUnpickler(protected val reader: TastyReader, nameAtRef: NameTable) {
       val expr = reader.ifBefore(end)(readTerm, EmptyTree)
       // TODO: always just taking the name?
       // return always returns from a method, i.e. something with a TermName
-      Return(expr, Ident(from.name.asInstanceOf[TermName]))
+      Return(expr, TermRefTree(from.name.asInstanceOf[TermName], TermRef(NoPrefix, from)))
```

Comment:
I guess this could be tested on `test-sources/src/main/scala/simple_trees/Return.scala`